var/spack/repos/builtin/packages/spark/package.py

```diff
@@ -30,6 +33,12 @@ class Spark(Package):
     version('1.6.1', 'fcf4961649f15af1fea78c882e65b001')
     version('1.6.0', '2c28edc89ca0067e63e525c04f7b1d89')
 
+    def url_for_version(self, version):
+        url = "http://archive.apache.org/dist/spark/spark-{0}/spark-{0}-bin-{1}.tgz"
+        if self.spec.satisfies('@2.4.0: +hadoop'):
+            return url.format(version, 'hadoop2.7')
+        return url.format(version, 'without-hadoop')
```

Comment:
I see what you are doing here, but I am not sure it copes well with the current logic of the `version` directive.